### ISSUE TITLE
`no-empty-source` performance optimization

### ISSUE THREAD

**Issue by gucong3000** [stylelint/stylelint#2798]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug when lint CSS in HTML about `no-empty-source`

> Which rule, if any, is this issue related to?

`no-empty-source`

> What CSS is needed to reproduce this issue?

html file without css.
https://github.com/vuejs-templates/webpack/blob/master/template/index.html

> What stylelint configuration is needed to reproduce this issue?

```js
module.exports = {
  rules: {
    "no-empty-source": true,
  }
}
```

> Which version of stylelint are you using?

stylelint@8.0.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint "**/*.{vue,html,css,sss,less,scss,sass}" --custom-syntax postcss-html`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to HTML.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
index.html
 35:1  ×  Unexpected empty source   no-empty-source
```
